src/components/graph/graph.helper.js

```diff
@@ -137,13 +137,15 @@ function _initializeNodes(graphNodes) {
  * @memberof Graph/helper
  */
 function _mapDataLinkToD3Link(link, index, d3Links = [], config, state = {}) {
-    const d3Link = d3Links[index];
+    // find the matching link if it exists
+    const d3Link = d3Links.find(l => l.source.id === link.source && l.target.id === link.target);
     const customProps = utils.pick(link, LINK_CUSTOM_PROPS_WHITELIST);
 
     if (d3Link) {
         const toggledDirected = state.config && state.config.directed && config.directed !== state.config.directed;
         const refinedD3Link = {
             ...d3Link,
+            index,
```

Comment:
This is correct the index should be passed into the link object as it's done in line `171` when we return.

Reply:
Ah cool, thanks for taking a look! Sorry for splitting this into two commits - at first I was letting the update use the old index.